server/src/main/java/org/elasticsearch/search/DefaultSearchContext.java

```diff
@@ -291,7 +291,7 @@ static int determineMaximumNumberOfSlices(
         ToLongFunction<String> fieldCardinality
     ) {
         return executor instanceof ThreadPoolExecutor tpe
-            && tpe.getQueue().isEmpty()
+            && tpe.getQueue().size() <= tpe.getMaximumPoolSize()
```

Comment:
In my test, this has the same effect as removing the conditional. That feels good, because we can maximize the parallelism without losing the protection against search rejections. This obviously depends on what you are testing :)  I think I am going to leave it this way for now, and we will eventually iterate over it as we do more testing / collect more info.

Reply:
Sounds good :) thanks!